sympy/core/assumptions.py

```diff
@@ -196,13 +196,29 @@ class StdFactKB(FactKB):
     """
     rules = _assume_rules
 
-    def __init__(self, facts=None):
+    def __init__(self, facts=None, extra_facts=None):
```

Comment:
Yes, essentially.  It is a bit of a hack.  

Explanation: Symbol forces "cummutative=True" if the user doesn't specify.  In master, Symbol just updates the dict before passing it to SkdFactKB.  But I want to save what the user specified, so I pass them in here separately and combine them after saving a copy of `facts`.

Alternatives: perhaps StdFactKB could be responsible for ensuring cummutative is True (if not specified).  This would simplify this bit of code a lot (there would be no `extra_facts`).  But I don't understand FactKB and StdFactKB well enough to know, e.g., this pseudocode is desirable:

````
def __init__(self, facts=None):
     self._generator = facts.copy()
     is_commutative = fuzzy_bool(assumptions.get('commutative', True))
     facts['commutative'] = is_commutative
     ....
     self.deduce_all_facts(facts)
````

What do you think?

(FWIW, I tried simpliy dropping this "cummutative=True" stuff (from Symbol) but I got test failures.)